### PR TITLE
adding shared services level tests

### PR DESCRIPTION
**Pull request by polatengin**
initial tests for the shared services level (Level 2)

the following tests are implemented;

- TestSharedServicesResourceGroupsExists
- TestSharedServicesHasOneResourceGroupForSharedServices
- TestSharedServicesHasRecoveryServiceVault
- TestSharedServicesHasTwoResourceGroupForNetworkingHub

To run the tests, there is a helper bash script `run_tests.sh`

Arguments of the helper is the following;

- "LANDING_ZONES_FOLDER" -z --zones
- "CONFIG_FOLDER" -c --config
- "ENVIRONMENT" -e --environment
- "CREATE_ENV" -ce --create-env
- "DEBUG_FLAG" -d --debug

If you want to both _deploy the infrastructure_ AND _run the tests_, execute the following command;

```bash
./run_tests.sh -z ~/caf/caf-terraform-landingzones -c ~/caf/caf-terraform-landingzones-starter/configuration -e demo -ce -d
```

If you just want to run the tests against an existing infrastructure, execute the following command;

```bash
./run_tests.sh -e demo -d
```

### CODE REVIEW

File: tests/level2_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+func TestSharedServicesResourceGroupsExists(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	client, _ := azure.GetResourceGroupClientE(test.SubscriptionID)
+
+	var top int32 = 100
+
+	result, _ := client.List(context.Background(), "tagName eq 'level' and tagValue eq 'level2'", &top)
```
Comment (hattan) [inline tests/level2_test.go:21]: This should also check that landingzone : shared_services. We want to ensure that this test only factors in the shared services lz and not launchpad. So we should be adding the condition and update expected value.

File: tests/level2_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	rgList := result.Values()
+
+	expected := 4
```
Comment (hattan) [inline tests/level2_test.go:25]: See above. expected should be 3 for this configuration.

File: tests/level2_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	result, _ := client.List(context.Background(), "tagName eq 'landingzone' and tagValue eq 'shared_services'", &top)
+
+	rgList := result.Values()
+
+	rg := rgList[0]
+
+	exist := azure.RecoveryServicesVaultExists(t, fmt.Sprintf("%s-rsv-vaultre1", test.Prefix), *rg.Name, test.SubscriptionID)
+
+	assert.True(t, exist, "Resource Group count does not match")
```
Comment (hattan) [inline tests/level2_test.go:65]: This message should indicate that the Recovery Service Vault was not found.

File: tests/level2_test.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+	var top int32 = 100
+
+	result, _ := client.List(context.Background(), "tagName eq 'landingzone' and tagValue eq 'networking_hub'", &top)
+
+	rgList := result.Values()
+
+	expected := 2
+
+	assert.Equal(t, expected, len(rgList), "Resource Group count does not match")
```
Comment (hattan) [inline tests/level2_test.go:83]: Can we add a descriptor to indicate that the Networking Hub resource group count doesn't match?

**Review by hattan — CHANGES_REQUESTED** [review]
Please see comments for specific changes requested. 

In addition, it might be good to add a test that verifies that the two VNETs are in different regions per.

File: tests/level2_test.go (hunk 1 above)
Comment (polatengin) [reply tests/level2_test.go]: updated test, thanks for the feedback

File: tests/level2_test.go (hunk 3 above)
Comment (polatengin) [reply tests/level2_test.go]: updated test, thanks for the feedback

File: tests/level2_test.go (hunk 4 above)
Comment (polatengin) [reply tests/level2_test.go]: error message updated, thanks for the feedback

File: tests/level2_test.go (hunk 2 above)
Comment (polatengin) [reply tests/level2_test.go:30]: updated test, thanks for the feedback

**Review by hattan — APPROVED** [review]
lgtm

**Pull request merged**